src/main/java/seedu/address/logic/parser/AddressBookParser.java

```diff
@@ -46,47 +53,121 @@ public Command parseCommand(String userInput) throws ParseException {
 
         final String commandWord = matcher.group("commandWord");
         final String arguments = matcher.group("arguments");
-        switch (commandWord) {
 
-        case AddCommand.COMMAND_WORD:
+        CommandType CommandType = getCommandType(commandWord);
+
+        switch (CommandType) {
+
+        case ADD:
             return new AddCommandParser().parse(arguments);
 
-        case EditCommand.COMMAND_WORD:
+        case EDIT:
             return new EditCommandParser().parse(arguments);
 
-        case SelectCommand.COMMAND_WORD:
+        case SELECT:
             return new SelectCommandParser().parse(arguments);
 
-        case DeleteCommand.COMMAND_WORD:
+        case DEL:
             return new DeleteCommandParser().parse(arguments);
 
-        case ClearCommand.COMMAND_WORD:
+        case CLEAR:
             return new ClearCommand();
 
-        case FindCommand.COMMAND_WORD:
+        case FIND:
             return new FindCommandParser().parse(arguments);
 
-        case ListCommand.COMMAND_WORD:
+        case LIST:
             return new ListCommand();
 
-        case HistoryCommand.COMMAND_WORD:
+        case HISTORY:
             return new HistoryCommand();
 
-        case ExitCommand.COMMAND_WORD:
+        case EXIT:
             return new ExitCommand();
 
-        case HelpCommand.COMMAND_WORD:
+        case HELP:
             return new HelpCommand();
 
-        case UndoCommand.COMMAND_WORD:
+        case UNDO:
             return new UndoCommand();
 
-        case RedoCommand.COMMAND_WORD:
+        case REDO:
             return new RedoCommand();
 
         default:
             throw new ParseException(MESSAGE_UNKNOWN_COMMAND);
         }
     }
 
+    /**
+     * Searches the entire list of acceptable command words in each command and returns the enumerated value type.
+     * @param commandWord
+     * @return enumerated value for the switch statement to process
+     */
+
+    private CommandType getCommandType(String commandWord) {
```

Comment:
can you think of a better way to parse the command type? this looks cumbersome

Reply:
I was actually aiming for this method to open up possibilities of different ways of calling the command but ill see how i can improve it